### ISSUE TITLE
Unable to hide fields based on FileUpload value

### ISSUE THREAD

**Issue by freshleafmedia** [laravel-filament/filament#5019]
### Package

filament/forms

### Package Version

2.16.51

### Laravel Version

9.40.1

### Livewire Version

2.10.7

### PHP Version

8.1.12

### Problem description

I want to conditionally hide fields based on whether the FileUpload field has been populated or not. This doesn't appear to be working:

```php
FileUpload::make('file_name')
    ->reactive(),

TextInput::make('test')
    ->visible(fn (callable $get): bool => $get('file_name') !== []),
```

### Expected behavior

The `test` `TextInput` should change it's visibility based on the state of the `FileUpload`

### Steps to reproduce

- Go to the Test resource create view
- Populate the FileUpload
- See that the TextInput field appears, as expected
- Clear the FileUpload
- See that the TextInput remains visible

### Reproduction repository

https://github.com/freshleafmedia/filament-bug-repro

**Comment by freshleafmedia** [comment]
Related discussion: https://github.com/filamentphp/filament/discussions/5015

**Comment by danharrin** [comment]
Fixed by bbf2923e8e3de54aa1457d122a557056a09dbee3
